src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java

```diff
@@ -95,49 +103,43 @@ protected void checkConflicts(Unit u, List<DataFlowAbstraction> statements){
         }
     }
 
-    @Override
-    protected void flowThrough(FlowSet<DataFlowAbstraction> in, Unit u, FlowSet<DataFlowAbstraction> out) {
-        detectConflict(in, u);
-        FlowSet<DataFlowAbstraction> temp = new ArraySparseSet<>();
-
-        FlowSet<DataFlowAbstraction> killSet = new ArraySparseSet<>();
-        for(DataFlowAbstraction item : in) {
-            if (statementEquals(item, u)){
-                killSet.add(item);
-            }
-        }
-        in.difference(killSet, temp);
-        temp.union(gen(u, in), out);
-    }
-
-    private String getLocalName(Local local){
-        return local.getName().split("#")[0];
-    }
-
-    private boolean statementEquals(DataFlowAbstraction statement, Unit u){
-
+    /*
+     * Compares the equality between the name
+     * of the current statement and unit variables.
+     */
+    private boolean statementEquals(DataFlowAbstraction statement, Unit u) {
         String statementName = getLocalName(statement.getLocal());
         for (ValueBox local : u.getDefBoxes()) {
```

Comment:
@rafaelmotaalves @barbosamaatheus  It doesn't cause a problem, because at the time of analyzing, the current unit, uses only the defBox, the useBox was the way to insert in the abstraction and get the name of the array.

Reply:
This does not solve the problem because in this case it takes Def only from U and compares it with the statement that has already been added to the abstraction. We really have a problem using Use and Def box

Reply:
You're right, @rafaelmotaalves and @barbosamaatheus . It can generate false positive. You may need to implement arrays differently. For the OAObjectFields, it's necessary to use both, def and use box, because it's generated many units.

Reply:
Won't this cause the same errors at the OAObjectFields, maybe would be good to execute the same tests @barbosamaatheus is using to check this. @galilasmb

Reply:
Done. I am now using `getDefBoxes()`.
I needed to add separate interactions for `Local `and `JArrayRef `cases

Reply:
@rafaelmotaalves I tested these cases. The OAObjectFields is a little different, a call from an object: o.a, is divided in others Units:
Example:
Class o = new Class();
o.a = 3;

These units are generated in Jimple:

$stack0 = new Class;
$stack1 = **o. $stack0;** // **UseBox is required here**
**$stack1.a** = 3; // **DefBox is required here**

Reply:
Other example:

Class object = new Class();
object.b.a.a = object.b.b.a + 3;

Jimple:

$stack2 = new Class;
object = $stack2;

$stack3 = object.b;
$stack8 = $stack3.a;

$stack4 = object.b; //this used to object.b.b.a
$stack5 = $stack4.b; //this used to object.b.b.a
$stack6 = $stack5.a; //this used to object.b.b.a

$stack7 = $stack6 + 3;

**$stack8.a** = $stack7;  //This is the search key for object.b.a.a

//The result in:
        $stack8.a -> $stack8 = $stack3.a; -> $stack3 = object.b; (STOP) -> Result: object.b.a.a

        object.b.b.a do not has search key :D

Reply:
Other example:

Class object = new Class();
object.b.a.a = object.b.b.a + 3;

Jimple:

$stack2 = new Class;
object = $stack2;

$stack3 = object.b;
$stack8 = $stack3.a;

$stack4 = object.b; //this used to object.b.b.a
$stack5 = $stack4.b; //this used to object.b.b.a
$stack6 = $stack5.a; //this used to object.b.b.a

$stack7 = $stack6 + 3;

**$stack8.a** = $stack7; //This is the search **key** of the call object.b.a.a

//The search result is:
**$stack8.a** -> $stack8 = $stack3.a; -> $stack3 = object.b; (STOP) -> Result: object.b.a.a

object.b.b.a has no search key :D